src/main/java/io/vertx/core/logging/LoggerFactory.java

```diff
@@ -46,18 +46,27 @@ public static synchronized void initialise() {
     }
 
     if (className != null) {
-      ClassLoader loader = Thread.currentThread().getContextClassLoader();
+      ClassLoader tcccl = Thread.currentThread().getContextClassLoader();
       try {
-        Class<?> clz = loader.loadClass(className);
-        delegateFactory = (LogDelegateFactory) clz.newInstance();
-      } catch (Exception e) {
-        throw new IllegalArgumentException("Error instantiating transformer class \"" + className + "\"", e);
+        newDelegateFactory = loadClass(className, tcccl);
+      } catch (IllegalArgumentException e) {
```

Comment:
Why catching an IllegalArgumentException here? ClassNotFoundException should be fine, no?

Reply:
Sorry, got it (didn't read the `loadClass` method.